### PR TITLE
Documentation: Clearer compatibility listing

### LINKED ISSUE #1873: Improve error reporting around CRD/TPR handling
```
if strings.Contains(errstr, "Failed to list *v1.NetworkPolicy: the server could not find the requested resource") {
  log.Warn("Consider upgrading k8s to >=1.7 to enforce NetworkPolicy version 1")
```

### PR DESCRIPTION
**Pull request by raybejjani**
**Summary of changes**:
This is a start at recording all the various versions we need in order to provide specific functionality. It is intended to address #1873 but needs more data about other cases to be useful. It is incomplete and I/we need to fill in sections marked ?? at least

@tgraf aanm fixed up that particular message last week. Did you still mean to have the issue open? Is this PR too broad for this case?

@aanm Is there a listing for k8s NetworkPolicy API versions that's similar to 
https://kubernetes.io/docs/reference/workloads-18-19/ ? I couldn't find one.

Fixes: #1873 

```release-note
Documented kubernetes API Group usage in docs
cilium status returns which kubernetes API Groups are supported/used by the agent
```

### CODE REVIEW

File: Documentation/compatibility.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
++---------------------------------------------------------------------+
+| k8s Version | cilium NetworkPolicy Features/Abilities               |
++=============+=======================================================+
+| <=1.6       | * k8s NetworkPolicy extensions/v1beta1                |
+|             | * CiliumNetworkPolicy as a ThirdPartyResource         |
++-------------+-------------------------------------------------------+
+| 1.7         | * k8s NetworkPolicy extensions/v1beta1                |
+|             | * k8s NetworkPolicy networking.k8s.io/v1              |
+|             | Exclusively one of:                                   |
```
Comment (aanm) [inline Documentation/compatibility.rst]: Remove the `Exclusively one of` and add `CiliumNetworkPolicy as a CustomResourceDefinition ` on all k8s versions after 1.6

**Review by aanm — CHANGES_REQUESTED** [review]
>@aanm Is there a listing for k8s NetworkPolicy API versions that's similar to
>https://kubernetes.io/docs/reference/workloads-18-19/ ? I couldn't find one.

I'm not aware of any

File: Documentation/compatibility.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+|            |         | * Use k8s etcd instance or standalone |
++------------+---------+---------------------------------------+
+| consul     | ??      |  * Distributed lock ID allocation     |
+|            |         | * Use k8s etcd instance or standalone |
++------------+---------+---------------------------------------+
+
+kernels
+-------
+
+cilium requires kernel version ?? and newer.
```
Comment (tgraf) [inline Documentation/compatibility.rst]: Requirements are covered here:
http://docs.cilium.io/en/stable/install/#system-requirements

File: Documentation/compatibility.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+??
+
+KV-Stores
+---------
+
++------------+-------------------------------------------------+
+| KVStore    | Version | cilium Features/Abilities             |
++============+=================================================+
+| etcd       | >??     | * Lock-free ID allocation             |
```
Comment (tgraf) [inline Documentation/compatibility.rst]: You can skip this, I'll add this to the docs together with the addition of the new kvstore code. This code has not been merged yet.

**Review by tgraf — CHANGES_REQUESTED** [review]

**Comment by raybejjani** [comment]
ok, @tgraf How does this status output look? I expect we'd list TPR/CRD as well. I noticed we were printing with two different formatters/writers, so the order is now different, with IPs listed last.
@aanm Do you think this shouldn't go into `status`? perhaps `config`?

Separately, I switched to a simple k8s compatibility page.

```
vagrant@cilium-k8s-master:~/go/src/github.com/cilium/cilium$ cilium status | head
KVStore:            Ok   Etcd: https://192.168.33.11:2379 - (Leader) 3.2.7
ContainerRuntime:   Ok
Kubernetes:         Ok   OK
Kubernetes APIs:    networking.k8s.io/v1, extensions/v1beta1
Cilium:             Ok   OK
NodeMonitor:        Listening for events on 2 CPUs with 64x4096 of shared memory
```

**Comment by aanm** [comment]
```
vagrant@cilium-k8s-master:~/go/src/github.com/cilium/cilium$ cilium status | head
KVStore:            Ok   Etcd: https://192.168.33.11:2379 - (Leader) 3.2.7
ContainerRuntime:   Ok
Kubernetes:         Ok   OK
Kubernetes APIs:    networking.k8s.io/v1, extensions/v1beta1  -> You need to add something
 saying this is related with kubernetes network policy because extensions/v1beta1 is too broad
 for example, the daemonsets are also under extensions/v1beta1. Also add which cilium version
 we are listening for (cilium.io/v1 or cilium.io/v2)
Cilium:             Ok   OK
NodeMonitor:        Listening for events on 2 CPUs with 64x4096 of shared memory
```

Answering your question, the IPs are only being printed when the agent runs in debug mode :-)

**Comment by raybejjani** [comment]
Righto. This was far more involved than expected! I can break some of the fixups into separate PRs, skim the commits to see what they are.

The output now shows:
```
KVStore:            Ok   Etcd: https://192.168.33.11:2379 - (Leader) 3.2.7
ContainerRuntime:   Ok
Kubernetes:         Ok   OK
Kubernetes APIs:    ["cilium/v2::CiliumNetworkPolicy", "extensions/v1beta1::NetworkPolicy", "networking.k8s.io/v1::NetworkPolicy", "core/v1::Service", "core/v1::Endpoint", "extensions/v1beta1::Ingress", "core/v1::Node", "CustomResourceDefinition"]
Cilium:             Ok   OK
NodeMonitor:        Listening for events on 2 CPUs with 64x4096 of shared memory
```

and we add and remove things in that list to reflect the actual Informers we have running, or when we succeed/fail at using TPR/CRDs. I think those are all the API Groups but maybe there are more we use?

**Review by tgraf — APPROVED** [review]

**Pull request merged**